library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -189,36 +248,56 @@ Http::FilterDataStatus PlatformBridgeFilter::decodeData(Buffer::Instance& data,
     });
   }
 
-  return onData(data, end_stream, internal_buffer, platform_filter_.on_request_data);
+  return onData(data, end_stream, internal_buffer, &pending_request_headers_,
+                platform_filter_.on_request_data);
 }
 
-Http::FilterTrailersStatus PlatformBridgeFilter::decodeTrailers(Http::RequestTrailerMap& trailers) {
+Http::FilterDataStatus PlatformBridgeFilter::encodeData(Buffer::Instance& data, bool end_stream) {
   // Delegate to shared implementation for request and response path.
-  return onTrailers(trailers, platform_filter_.on_request_trailers);
-}
+  Buffer::Instance* internal_buffer = nullptr;
+  if (encoder_callbacks_->encodingBuffer()) {
+    encoder_callbacks_->modifyEncodingBuffer([&internal_buffer](Buffer::Instance& mutable_buffer) {
+      internal_buffer = &mutable_buffer;
+    });
+  }
 
-Http::FilterHeadersStatus PlatformBridgeFilter::encodeHeaders(Http::ResponseHeaderMap& headers,
-                                                              bool end_stream) {
-  // Delegate to shared implementation for request and response path.
-  return onHeaders(headers, end_stream, platform_filter_.on_response_headers);
+  return onData(data, end_stream, internal_buffer, &pending_response_headers_,
```

Comment:
Is passing a null buffer here acceptable? (I assume yes)

Reply:
Yes, but good eye for catching the possibility. Here, `nullptr` represents "none".